code/src/sixsq/nuvla/server/resources/spec/voucher.cljc

```diff
@@ -0,0 +1,189 @@
+                                       :default "new"})))
+
+
+(s/def ::service-info
+  (-> (st/spec ::cimi-core/url)
+      (assoc :name "service-info"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "service-info"
+             :json-schema/description "URL for the service provider who issued the voucher"
+             :json-schema/group "body"
+             :json-schema/order 24
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::code
+  (-> (st/spec ::cimi-core/nonblank-string)
+      (assoc :name "code"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "code"
+             :json-schema/description "voucher code"
+             :json-schema/group "body"
+             :json-schema/order 25
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::activated
+  (-> (st/spec ::cimi-core/timestamp)
+      (assoc :name "activated"
+             :json-schema/type "date-time"
+             :json-schema/editable true
+
+             :json-schema/display-name "activated"
+             :json-schema/description "when the voucher was activated"
+             :json-schema/group "body"
+             :json-schema/order 26
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::redeemed
+  (-> (st/spec ::cimi-core/timestamp)
+      (assoc :name "redeemed"
+             :json-schema/type "date-time"
+             :json-schema/editable true
+
+             :json-schema/display-name "redeemed"
+             :json-schema/description "when the voucher was redeemed"
+             :json-schema/group "body"
+             :json-schema/order 27
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::target-audience
+  (-> (st/spec ::cimi-core/nonblank-string)
+      (assoc :name "target-audience"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "target-audience"
+             :json-schema/description "who the voucher is targeted for"
+             :json-schema/group "body"
+             :json-schema/order 28
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+
+(s/def ::batch-reference
+  (-> (st/spec ::cimi-core/nonblank-string)
+      (assoc :name "batch-reference"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "batch-reference"
+             :json-schema/description "reference for the batch this voucher belongs to"
+             :json-schema/group "body"
+             :json-schema/order 29
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::wave-id
```

Comment:
Probably `wave` is sufficient.  In any case, avoid `wave-id` unless this is an identifier that points to another resource on the API server.

Reply:
done